### ISSUE TITLE
Cannot use boolean values in conditional configuration

### ISSUE THREAD

**Issue by MiYanni** [Azure/autorest#3442]
In configuration, you can use a condition in the `yaml` blocks. It looks like this:
````
``` yaml $(example-value)
# Do stuff here
```
````

However, the injection of the `$(example-value)` doesn't work with booleans. For example:
````
``` yaml
example-value: false
```

``` yaml $(example-value)
# Do stuff here
```
````

In this situation, the second block executes. Why? Because it does object evaluation instead of boolean evaluation. Meaning, setting `example-value` to anything will result in a truthy statement. The only way to explicitly make `example-value` a falsey statement is to set it to `null`. It will also default to a falsey statement because there is no value set by default. So:
````
``` yaml
example-value: false # this is truthy
example-value: true # this is truthy
example-value: 1337 # this is truthy
example-value: 'fasdkjfjdsalkjlk' # this is truthy
example-value: null # this is falsey
```

``` yaml $(example-value)
# Do stuff here
```
````

Basically, this issue is to make booleans act like booleans so `true` is truthy, and `false` is falsey.

**Comment by fearthecowboy** [comment]
That's not expected. `false` value should be a negative condition.

**Comment by timotheeguerin** [comment]
Believe this is fixed now.
